test/parallel/test-net-write-after-close.js

```diff
@@ -21,24 +21,27 @@
 
 'use strict';
 const common = require('../common');
+
 const net = require('net');
 
+let serverSocket;
+
 const server = net.createServer(common.mustCall(function(socket) {
+  serverSocket = socket;
+
   socket.resume();
 
   socket.on('error', common.mustCall(function(error) {
     console.error('got error, closing server', error);
     server.close();
   }));
-
-  setTimeout(common.mustCall(function() {
-    console.error('about to try to write');
-    socket.write('test', common.mustCall());
-  }), 250);
 }));
 
 server.listen(0, function() {
   const client = net.connect(this.address().port, function() {
+    client.on('end', () => {
```

Comment:
Should documentation for the `finish` event be added to [`net.Socket` documentation](https://nodejs.org/api/net.html#net_class_net_socket the way) the way `end` is? Or would that just be more confusing than helpful?